acceptancetests/README.md

```diff
@@ -1,3 +1,136 @@
+# Juju Functional Tests
+
+## Running Test Scripts 
+
+Some setup is required before running the tests.
+
+### Installing dependencies
+
+Dependencies are installed via a .deb and pip install.
+
+```bash
+$ sudo apt-get install make
+
+$ make install-deps
+```
+
+### Required Environment Variables
+
+  * **JUJU_HOME**: The directory the test will use to:
+     * Find environments.yaml [(see use of environments.yaml below)](#envs)
+     * Find credentials.yaml [(see use of credentials.yaml below)](#envs-creds)     
+     * Create a **JUJU_DATA** dir for use for the duration of the test. The directory is created in: $JUJU_HOME/juju-homes/.
+  * **JUJU_REPOSITORY**: The directory containing the local dummy charms. You can use '\<juju root\>/acceptancetests/repository'.
+
+
+### Quick run using LXD
+To run a test locally with lxd:
+
+  * ```$ mkdir /tmp/test-run```
+  * ```$ export JUJU_HOME=/tmp/test-run```
+  * ```$ vim $JUJU_HOME/environments.yaml```
+    ```yaml
+    environments:
+        lxd:
+            type: lxd
+            test-mode: true
+            default-series: xenial
+    ```
+  * ```export JUJU_REPOSITORY=./path/to/acceptancetests/repository```
+  * Now you can run the test with:
+     * ```$ ./assess_model_migration.py lxd . . .```
+
+### Quick run using AWS
+
+See [(Use of environments.yaml below)](#envs) and [(Use of credentials.yaml below)](#envs-creds) for a full explanation of the files used here.
+
+To run a test using AWS:
+
+  * ```$ mkdir /tmp/test-run```
+  * ```$ export JUJU_HOME=/tmp/test-run```
+  * ```$ vim $JUJU_HOME/environments.yaml```
+    ```yaml
+    environments:
+        myaws:
+            type: ec2
+            test-mode: true
+            default-series: xenial
+            region: us-east-1
+    ```
+  * ```$ vim $JUJU_HOME/credentials.yaml```
+    ```yaml
+    credentials:
+      aws:
+        credentials:
+          auth-type: access-key
+          access-key: <access key>
+          secret-key: <secret key>
+    ```
+  * ```export JUJU_REPOSITORY=/path/to/acceptancetests/repository```
+  * Now you can run the test with:
+     * ```$ ./assess_model_migration.py myaws . . .```
+
+### Which juju binary is used?
+
+If no *juju_bin* argument is passed to an *assess* script it will default to using the juju in your **$PATH**.
+
+So, to use a locally build juju binary either:
+
+  * Ensure you binary is in **$PATH** (i.e. export PATH=/home/user/src/Go/bin:$PATH)
+  * Or explicitly pass it to the *assess* script: ./assess_something.py lxd  /home/user/src/Go/bin/juju
+
+### Using an existing controller
```

Comment:
I think it's worth mentioning how to use this to run a test. Could you add a section perhaps telling people to bootstrap a controller, then run with --existing? Or is that clear already?